esp-hal/src/timer/systimer.rs

```diff
@@ -985,26 +814,20 @@ mod asynch {
     static WAKERS: [AtomicWaker; NUM_ALARMS] = [const { AtomicWaker::new() }; NUM_ALARMS];
 
     #[must_use = "futures do nothing unless you `.await` or poll them"]
-    pub(crate) struct AlarmFuture<'a, COMP: Comparator, UNIT: Unit> {
-        alarm: &'a Alarm<'a, Target, crate::Async, COMP, UNIT>,
+    pub(crate) struct AlarmFuture<'a> {
+        alarm: &'a Alarm<'a>,
     }
 
-    impl<'a, COMP: Comparator, UNIT: Unit> AlarmFuture<'a, COMP, UNIT> {
-        pub(crate) fn new(alarm: &'a Alarm<'a, Target, crate::Async, COMP, UNIT>) -> Self {
+    impl<'a> AlarmFuture<'a> {
+        pub(crate) fn new(alarm: &'a Alarm<'a>) -> Self {
             alarm.clear_interrupt();
 
-            let (interrupt, handler) = match alarm.comparator.channel() {
-                0 => (Interrupt::SYSTIMER_TARGET0, target0_handler),
-                1 => (Interrupt::SYSTIMER_TARGET1, target1_handler),
-                _ => (Interrupt::SYSTIMER_TARGET2, target2_handler),
-            };
-
-            unsafe {
-                interrupt::bind_interrupt(interrupt, handler.handler());
-                interrupt::enable(interrupt, handler.priority()).unwrap();
-            }
-
-            alarm.set_interrupt_handler(handler);
+            alarm.set_interrupt_handler(match alarm.comparator.channel() {
```

Comment:
This is unfortunately problematic because we can overwrite the user's interrupt handler without notice. If we offer an async API OR configurable interrupt handlers, we should have the Async/Blocking distinction.

Reply:
I've removed it, it was leftover code. The low-level future shouldn't be setting the interrupt handler at all.